src/figuresClasses.ts

```diff
@@ -1,19 +1,85 @@
-export interface Figure {
+enum Shape {
+  triangle = 'triangle',
+  rectangle = 'rectangle',
+  circle = 'circle'
+}
+enum Color {
+  red = 'red',
+  green = 'green',
+  blue = 'blue',
+}
 
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  shape = Shape.triangle;
 
-}
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (this.a <= 0 || this.b <= 0 || this.c <= 0) {
+      throw new Error('Incorrect value');
+    }
+
+    const max: number = Math.max(this.a, this.b, this.c);
 
-export class Circle {
+    if (max >= (this.a + this.b + this.c - max)) {
+      throw new Error('Sides of triangle not correct');
+    }
+  }
 
+  getArea(): number {
+    const p: number = (this.a + this.b + this.c) / 2;
+    const s: number = Math.sqrt(p * (p - this.a) * (p - this.b) * (p - this.c));
```

Comment:
use more readable names for variables, e.g. `area`
you can use a, b, c for sides it's ok, but for area and perimeter better use full names
and check it in whole code